### ISSUE TITLE
Kube-scheduler: invalid memory address in podFitsOnNode() caused crash

### ISSUE THREAD

**Issue by zhan849** [kubernetes/kubernetes#92050]
**What happened**:
kube scheduler has NPE in `podFitsOnNode`

```bash
E0611 18:26:25.851090       8 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 7717 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1812d80, 0x2b2ae40)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1812d80, 0x2b2ae40)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/pkg/scheduler/core.(*genericScheduler).podFitsOnNode(0xc0008ca240, 0xc038cfd2a0, 0xc0172ea800, 0x1d0ece0, 0xc03833f9e0, 0x0, 0xc0008c4060, 0x1d52ba0, 0xc00069e0b0, 0xc0180edf00, ...)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/core/generic_scheduler.go:676 +0x1a1
k8s.io/kubernetes/pkg/scheduler/core.(*genericScheduler).findNodesThatFit.func1(0x3a8)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/core/generic_scheduler.go:492 +0x1b1
k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.ParallelizeUntil.func1(0xc03c3308f4, 0xc038c39000, 0xc001d54340, 0xc01f4e8850)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:57 +0xbf
created by k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.ParallelizeUntil
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:49 +0x148
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1614f51]

goroutine 7717 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1812d80, 0x2b2ae40)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/pkg/scheduler/core.(*genericScheduler).podFitsOnNode(0xc0008ca240, 0xc038cfd2a0, 0xc0172ea800, 0x1d0ece0, 0xc03833f9e0, 0x0, 0xc0008c4060, 0x1d52ba0, 0xc00069e0b0, 0xc0180edf00, ...)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/core/generic_scheduler.go:676 +0x1a1
k8s.io/kubernetes/pkg/scheduler/core.(*genericScheduler).findNodesThatFit.func1(0x3a8)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/scheduler/core/generic_scheduler.go:492 +0x1b1
k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.ParallelizeUntil.func1(0xc03c3308f4, 0xc038c39000, 0xc001d54340, 0xc01f4e8850)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:57 +0xbf
created by k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue.ParallelizeUntil
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/util/workqueue/parallelizer.go:49 +0x148
```
seems nodeInfo is nil here

**What you expected to happen**:
return proper error when nodeInfo is nil

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.16.8
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/sig scheduling

**Comment by Huang-Wei** [comment]
@zhan849 which k8s version are you running on?

**Comment by zhan849** [comment]
Hi @Huang-Wei , it's 1.16.8

**Comment by Huang-Wei** [comment]
@alculquicondor I recalled there is a similar NPE issue you fixed recently? Could you link that here, and then let's check if it's backported to 1.16.8.

**Comment by alculquicondor** [comment]
/close

dup of #89006

Fixed in 1.16.9 https://github.com/kubernetes/kubernetes/compare/v1.16.8...v1.16.9
